training_main.py

```diff
@@ -0,0 +1,354 @@
+
+all_desc=set()
+for key in descriptions.keys():
+    #print(key)
+    for d in descriptions[key]:
+        [all_desc.update(d.split())]
+
+# %%
+
+# Printing the number of distinct words present in the descriptions
+        
+vocabulary=all_desc
+print("size of the vocabulary=",len(vocabulary))
+
+# %%
+
+# Loading and printing the size of the description dataset
+
+filename='results.csv'
+train=description_processor.load_set(filename)
+print('dataset=',len(train))
+
+# %%
+
+# Initializing the images-dataset directory and adding the name of each image into 'img' list
+
+images='./flickr30k_images/flickr30k_images/'
+img=glob.glob(images+'*.jpg')
+
+# %%
+
+# Initializing and customizing the InceptionV3 model
+
+model=InceptionV3(weights='imagenet')
+model_new=Model(model.input,model.layers[-2].output)
+
+# %%
+
+# Setting time-stamp as current time and initializing empty dictionary 'encoding_train'
+start=time()
+encoding_train={}
+
+# Looping through and encoding all the images in the train directory
+
+for img in img:
+    encoding_train[img[len(images):]]=image_processing.encode(img)
+    print("Time taken in second=",time()-start)
+
+# %%
+
+# Pickling the encoded training images dataset 
+    
+with open('./resources/encoded_train_images.pkl','wb') as f:
+    dump(encoding_train,f)
+
+# %%
+
+# Loading the pickled encoded-image dataset and initializing it as the training feature matrix
+    
+train_features=load(open('./resources/encoded_train_images.pkl','rb'))
+print(len(train_features))
+
+# %%
+
+# Adding all the training captions to a list and printing it's length
+
+all_train_captions=[]
+for key,val in descriptions.items():
+    for cap in val:
+        all_train_captions.append(cap)
+len(all_train_captions)
+
+# %%
+
+# Prints the word count of the description dataset and initializes the frequent-words list
+
+vocab=description_properties.description_vocabulary(all_train_captions)
+
+# %%
+
+# Creating dictionaries mapping indices to words and vice-versa
+
+ixtoword={}
+wordtoix={}
+ix=1
+for w in vocab:
+    wordtoix[w]=ix
+    ixtoword[ix]=w
+    ix+=1
+    
+# %%
+
+# Saving the mapping dictionaries locally
+    
+with open('./resources/ixtoword.pkl','wb') as f1:
+    dump(ixtoword,f1)
+    f1.close()
+
+with open('./resources/wordtoix.pkl','wb') as f2:
+    dump(wordtoix,f2)
+    f2.close()
+
+# %%
+
+# Initializing and printing the size of the vocabulary using the index-to-word mapping dict
+vocab_size=len(ixtoword)+1
+print(vocab_size)
+
+# %%
+
+# Prints the number of lines of descriptions in the training set
+
+print(len(description_properties.to_lines(descriptions)))
+
+# %%
+
+# Saves and prints the length of the longest description in the description dataset
+
+max_length=description_properties.max_length(descriptions)
+print("Length of max-length description = ",max_length)
+
+# %%
+
+# Importing the 'glove' word-set and storing it in 'embedding_index'
+
```

Comment:
loading of glove vectors and making of the embedding matrix shoule be done in other file and only the function should be called here to get the embedding matrix